stl/inc/xutility

```diff
@@ -4123,24 +4123,14 @@ public:
         return _Left._Current - _Right._Get_last();
     }
 
-    _NODISCARD friend constexpr reference iter_move(const move_iterator& _It)
-#ifdef __EDG__ // TRANSITION, VSO-1132105

```

Comment:
Could you please leave these in place? The fix for VSO-1132105 isn't sufficient for these occurrences (or those in #1287). I apparently overreduced the repro, and these still cause errors in internal tests with `__cpp_lib_concepts` defined. I *think* removing the occurrences in `<functional>` is fine since they refer to members declared before the function with the `noexcept`-specifier. These occurrences fail because `_Current` is declared later in the class and apparently EDG didn't get the memo that `noexcept`-specifiers are a complete-class context. (The problem that #1287 is fixing in `<ranges>` is similar.)

I need to file a new bug, and I'll re-annoate these in #1287.

Reply:
(For clarity: this is the actual requested change, the others are merely suggestions.)

Reply:
Thanks - it feels like we've been encountering some form of this bug for almost a decade now. I will revert this part.